Vector.java

```diff
@@ -0,0 +1,41 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class Vector {
+	
+	List<Double> numbers;
+		
+	Vector (List<Double> numbers) {
+			
+		numbers = numbers;
+			
+	}
+		
+	public double distance(Vector vector2) { // returns distance between two vectors
+		
+		double squared_distance = 0;
+			
+		for (int index = 0; index < numbers.size(); index++) {
+			double difference = numbers.get(index) - vector2.numbers.get(index);
+			squared_distance += (difference * difference);
+		}
+			
+		return Math.sqrt(squared_distance);	
+					
+	}
+	
+	public double magnitude() { // computes vector magnitude
```

Comment:
Your implementation is correct, but its needlessly expensive.  Yes, the magnitude of a vector is the distance from the origin, and it's nice that you were able to use the helper function, but in this implementation, you are looping once to build up a list, creating a new instance of the Vector class, and will loop again in the call to `distance`.  This function should just be a single loop that accumulates a running sum of squares.